### PR TITLE
Reduced the number of channels between service and process

### PR DESCRIPTION
**Pull request by ysingh7**
Issue Number: Remove hacks and refactor Runtime and RuntimeService/Synchronizer #86

Objective of pull request: Reduced the number of channels between service and process

## Pull request checklist

Your PR fulfills the following requirements:
- [X ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [X ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [X] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [ ] Lint (`pyb`) passes locally
- [ ] Build tests (`pyb -E unit`) or (`python -m unittest`) passes locally

## Pull request type

 

 

Please check your PR type:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

- Currently we have five channels between runtime service and process model. 
- 1. Used for sending cmd from RS to PM
- 2. Used for sending ack back from PM to RS
- 3. Used for sending READ/WRITE Req from RS to PM
- 4. Used to send write data from RS to PM
- 5. Used to send read data from PM to RS

## What is the new behavior?

- Reduces the number of channels between runtime service and process model. Also it removes the notion of time step from the runtime.
- 1. One way channel for all cmd/get-set-req/data from RS to PM
- 2. One channel for all ack/data from PM to RS

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: tests/lava/magma/runtime/test_get_set_var.py [hunk 1]
```diff
@@ -92,7 +92,6 @@ def test_get_set_var_using_runtime(self):
                               expected_result_u)
         assert np.array_equal(process._runtime.get_var(process.v.id),
                               expected_result_v)
-        self.assertEqual(process.runtime.global_time, 15)
```
Comment (PhilippPlank) [inline tests/lava/magma/runtime/test_get_set_var.py:95]: We should still be able to get the time step out of the process, right?

File: tests/lava/magma/runtime/test_get_set_var.py [hunk 2]
```diff
@@ -129,7 +128,6 @@ def test_get_set_var_using_var_api(self):
         process.run(condition=RunSteps(num_steps=5), run_cfg=run_config)
         assert np.array_equal(process.u.get(), expected_result_u)
         assert np.array_equal(process.v.get(), expected_result_v)
-        self.assertEqual(process.runtime.global_time, 15)
```
Comment (PhilippPlank) [inline tests/lava/magma/runtime/test_get_set_var.py:132]: Same as above, compare with the time step in ProcModel

File: tests/lava/magma/runtime/test_loihi_protocol.py [hunk 3]
```diff
@@ -59,7 +59,6 @@ def test_synchronization_single_process_model(self):
         run_config = SimpleRunConfig(sync_domains=[simple_sync_domain])
         process.run(condition=RunSteps(num_steps=10), run_cfg=run_config)
         process.run(condition=RunSteps(num_steps=5), run_cfg=run_config)
-        self.assertEqual(process.runtime.global_time, 15)
```
Comment (PhilippPlank) [inline tests/lava/magma/runtime/test_loihi_protocol.py:62]: Same

File: src/lava/magma/runtime/runtime.py [hunk 4]
```diff
@@ -44,17 +45,15 @@ class Runtime:
     be blocking and non-blocking as specified by the run run_condition."""
 
     def __init__(self,
-                 run_cond: AbstractRunCondition,
                  exe: Executable,
                  message_infrastructure_type: ActorType):
-        self._run_cond: AbstractRunCondition = run_cond
+        self._run_cond: typing.Optional[AbstractRunCondition] = None
```
Comment (PhilippPlank) [inline src/lava/magma/runtime/runtime.py:50]: Why is the RunCondition optional? This internal RunCondition is and was not used by the Runtime, as far as I can see. I think we do not need it, correct?

**Review by PhilippPlank — APPROVED** [review]
Good start I think, please look at my comments.

**Review by joyeshmishra — APPROVED** [review]

**Pull request merged**